Automattic/woocommerce-payments
# Split UPE: Correct token behavior

Fixes #4530

#### Changes proposed in this Pull Request

This PR ensures that tokens use the correct gateway ID and are displayed under the correct gateway throughout WCPay. Some checks have also been added to ensure that tokens aren't duplicated per each gateway.









#### Testing instructions





* Enable SEPA in the UPE settings
* From My Account > Payment Methods > Add Payment Method, add a credit card and SEPA method
* Return to My Account > Payment Methods > Add Payment Method and refresh the page a few times to make sure tokens aren't being duplicated
* Check that SEPA tokens are appearing under SEPA Direct Debit and card tokens are appearing under Credit card / Debit card

-------------------


**Post merge**
